baseplate/frameworks/pyramid/__init__.py

```diff
@@ -294,3 +299,20 @@ def pshell_setup(env: Dict[str, Any]) -> None:
 
     """
     env["request"].start_server_span("shell")
+
+
+def get_is_healthy_probe(request: Request) -> int:
+    """Get the thrift enum value of the probe used in http is_healthy request."""
+    code = request.params.get("type", str(IsHealthyProbe.READINESS))
+    try:
+        return int(code)
+    except ValueError:
+        pass
+    # If it's not an int, try to find it in the enum map instead.
+    try:
+        return IsHealthyProbe._NAMES_TO_VALUES[code.upper()]
+    except KeyError:
+        logger.warning(
+            "Unrecognized health check type %s, fallback to READINESS", code,
+        )
+        return IsHealthyProbe.READINESS
```

Comment:
this feels like the sorta thing we should just leave up to applications. if we bake it in here they have no choice on how strict to be.

Reply:
The design doc explicitly stated that for any unrecognized probes the service should always fallback to READINESS.

By definition READINESS is the most strict probe. It's also (essentially) the only one we actually have in the original version of health check.